src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopy = { ...state };
+  const versionsOfState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopy, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const KeyToRemove of action.keysToRemove) {
+          delete stateCopy[KeyToRemove];
+        }
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          if (key) {
+            delete stateCopy[key];
+          }
+        }
```

Comment:
It's just copy, so we can simplify it

```suggestion
        stateCopy = {}
```